draft-sharabayko-mops-srt.md

```diff
@@ -2086,15 +2086,583 @@ and RTTVar are updated by the socket based on algorithms for the sender
+periodic NAK reports ({{packet-naks}}) is updated as follows:
+
+~~~
+NAKInterval = min((RTT + 4 * RTTVar) / 2, 20000)
+~~~
+
+where RTT and RTTVar are receiver's estimations (see {{ctrl-pkt-ack}},
+{{rtt}}). The minimum value of NAKInterval is set to 20 milliseconds
+in order to avoid sending periodic NAK reports too often under
+low latency conditions.
+
+## File Transfer Congestion Control (FileCC)
+
+For file transfer ({{transmission-mode-msg}},
+{{transmission-mode-buffer}}), any known file congestion control
+algorithm like CUBIC {{RFC8312}} or BBR {{BBR}} can be applied,
+including SRT's default FileCC algorithm decsribed below
+({{default-fileCC}}).
+
+### SRT's Default FileCC Algorithm {#default-fileCC}
+
+SRT's default FileCC algorithm is a modified version of the UDT's
+native congestion control algorithm {{GuAnAO}}, {{GHG04b}}
+designed for a bulk data transfer over networks with a large
+bandwidth-delay product (BDP). It's a hybrid Additive Increase
+Multiplicative Decrease (AIMD) algorithm, hence it adjusts both
+congestion window size (CWND_SIZE) and packet sending period
+(PKT_SND_PERIOD). The units of measurement for CWND_SIZE and
+PKT_SND_PERIOD are packets and microseconds, respectively.
+
+The algorithm controls sending rate by tuning packet sending period,
+i.e. how often to send out the packets. The sending rate is increased
+upon an acknowledgement (ACK) reception and decreased in case of
+receiving a loss report (negative acknowledgement, or NAK).
+Only full ACKs (not light) trigger the increase of sending rate.
+
+The congestion control includes two phases. It starts with a slow
+start phase where it probes the network to determine available
+bandwidth and the target sending rate for the desired operation
+mode: congestion avoidance. In this phase, if there is no congestion
+detected via loss reports, the sending rate is gradually increased.
+On the other hand, if a network congestion is detected, the algorithm
+slows down and decreases the sending rate to reduce following packet
+loss. The slow start phase runs exactly once at the beginning of a
+connection and stops when a packet loss occurs, congestion window
+size reaches its maximum value, or on a timeout event.
+
+The detailed algorithm behaviour at both phases is described
+in {{default-fileCC-slow-start}} and
+{{default-fileCC-congestion-avoidance}}, respectively.
+
+#### Slow Start {#default-fileCC-slow-start}
+
+During the slow start phase, the packet sending period PKT_SND_PERIOD
+is kept at 1 microsecond in order to send packets as fast as possible,
+but not with an infinite rate.
+The initial value of congestion window size CWND_SIZE is set to 16
+packets. CWND_SIZE has an upper threshold, the maximum allowed
+congestion window size MAX_CWND_SIZE, which is used for safety purposes
+(even if there is no packet loss, the slow start phase has to stop
+at a certain point). The threshold can be set to the maximum of
+the receiver buffer size (12 MB).
```

Comment:
```suggestion
The initial value of the congestion window size (CWND_SIZE) is set to 16
packets. CWND_SIZE has an upper threshold, which is the maximum allowed
congestion window size (MAX_CWND_SIZE), so that even if there is no packet 
loss, the slow start phase has to stop at a certain point. The threshold can be 
set to the maximum receiver buffer size (12 MB).
```